### DISCUSSION TITLE
Why is GitHub running my code?

### DISCUSSION THREAD

**Discussion by NYUeraLabs** [community/community#50203] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi,
I am constantly getting an email saying that one of my private repos failed to run. Can someone tell me why?

I have never run anything from GitHub, I only keep my repos there. I have never given anyone permission to do so (including GitHub).

My main concern is that it shows billing time. And the fact that a private repo is being run without my permission.

Is our code safe on GitHub? 
![Screenshot_20230316_100306_com android chrome](https://user-images.githubusercontent.com/114792993/225554709-1edb792c-f7c6-448b-b3b5-e2942f1eb9eb.jpg)

Please check screenshot attached.

**Comment by airtower-luna** [comment]
The screenshot looks like your repository contains a workflow for GitHub Actions, and a someone complex one at that, so you did actually tell GitHub to run certain things. If you don't want that, you can remove the workflow(s) or disable Actions in the repository settings.

You get a certain number of minutes for free, for more you'd actually need to pay, see: https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions
